### DISCUSSION TITLE
How to get ancestors of an entry by slug in graphQL?

### DISCUSSION THREAD

**Discussion by ctigelaar** [craftcms/cms#8013] [Q&A] [answered]
I'm using a structure with nesting several levels deep. For the breadcrumbs I want to get a path (array of ancestors) to the current entry, based on the "slug". In Twig there seems a way with `getAncestors` (https://docs.craftcms.com/api/v3/craft-base-elementinterface.html#method-getancestors), but in GraphQL I couldn't find a way to get this done. Is it possible in 1 query? The `ancestorsOf` doesn't take a `slug`, and at that point I don't have the `id` of the requested entry. Would be nice to just read them from an `ancestors` property in the `entry`. But maybe I'm missing something and it's already possible?

```
{
  entry(section: "foobar", slug: "my-random-slug") {
    id
    title
  }
  parents: entries(section: "foobar", ancestorOf: 2800) { // id is not available at this moment?
    id
    title
  }
}
```
PS. I found some proposed solutions with `compose` or something, but I'm trying to keep as close as possible to standard behaviour here.

**Comment by brandonkelly** [comment, marked as answer]
We’ve actually [just added](https://github.com/craftcms/cms/commit/68eb7c7726a2033459e449de84ef227937d59447) the ability to fetch entries’ ancestors for Craft 3.7.

To get on the 3.7 branch, change your `craftcms/cms` requirement in composer.json to `"3.7.x-dev as 3.7.0-alpha"` and run `composer update`.

**Reply by ctigelaar** [reply]
Perfect, just in time for me then! Thanks a lot for the effort put into improving Craft, much appreciated.
